### ISSUE TITLE
Rootless podman exec fails to exec and remove

### ISSUE THREAD

**Issue by frezbo** [containers/libpod#1337]
**Shell o/p**

```
~ (aws:rean-gov-sd)(kc)$ podman image ls
REPOSITORY                           TAG      IMAGE ID       CREATED       SIZE
registry.access.redhat.com/rhel7.5   latest   f5ea21241da8   2 weeks ago   211MB
docker.io/library/alpine             latest   11cd0b38bc3c   7 weeks ago   4.67MB
~ (aws:rean-gov-sd)(kc)$ podman run -d registry.access.redhat.com/rhel7.5 sleep 100
b51f7325456e06bfdd13f45228a6c0175ce4228682b685f2352e7cf22f0c6c10
~ (aws:rean-gov-sd)(kc)$ podman exec -it b51f7325456e06bfdd13f45228a6c0175ce4228682b685f2352e7cf22f0c6c10 /bin/bash
exec failed: container_linux.go:336: starting container process caused "process_linux.go:86: executing setns process caused \"exit status 21\""
exit status 1
~ (aws:rean-gov-sd)(kc)$ podamn rm -f b51f7325456e06bfdd13f45228a6c0175ce4228682b685f2352e7cf22f0c6c10
bash: podamn: command not found...
Similar command is: 'podman'
~ (aws:rean-gov-sd)(kc)$ podman rm -f b51f7325456e06bfdd13f45228a6c0175ce4228682b685f2352e7cf22f0c6c10
failed to delete container b51f7325456e06bfdd13f45228a6c0175ce4228682b685f2352e7cf22f0c6c10: cannot remove container b51f7325456e06bfdd13f45228a6c0175ce4228682b685f2352e7cf22f0c6c10 as it could not be stopped: container b51f7325456e06bfdd13f45228a6c0175ce4228682b685f2352e7cf22f0c6c10 did not die within timeout
~ (aws:rean-gov-sd)(kc)$ runc list
ID                                                                 PID         STATUS      BUNDLE                                                                                                                                  CREATED                          OWNER
b51f7325456e06bfdd13f45228a6c0175ce4228682b685f2352e7cf22f0c6c10   14304       running     /home/frezbo/.local/share/containers/storage/vfs-containers/b51f7325456e06bfdd13f45228a6c0175ce4228682b685f2352e7cf22f0c6c10/userdata   2018-08-24T14:21:01.643048878Z   frezbo
~ (aws:rean-gov-sd)(kc)$ runc kill b51f7325456e06bfdd13f45228a6c0175ce4228682b685f2352e7cf22f0c6c10 KILL
~ (aws:rean-gov-sd)(kc)$ podman rm -f b51f7325456e06bfdd13f45228a6c0175ce4228682b685f2352e7cf22f0c6c10
b51f7325456e06bfdd13f45228a6c0175ce4228682b685f2352e7cf22f0c6c10
~ (aws:rean-gov-sd)(kc)$ podman run -d docker.io/library/alpine sleep 100
46490b92bc2ab1527579a66a2da8ba04192acf629d7f168f19cabbb0fb2181ac
~ (aws:rean-gov-sd)(kc)$ podman exec -it 46490b92bc2ab1527579a66a2da8ba04192acf629d7f168f19cabbb0fb2181ac /bin/sh
exec failed: container_linux.go:336: starting container process caused "process_linux.go:86: executing setns process caused \"exit status 21\""
exit status 1
~ (aws:rean-gov-sd)(kc)$ podman rm -f 46490b92bc2ab1527579a66a2da8ba04192acf629d7f168f19cabbb0fb2181ac
failed to delete container 46490b92bc2ab1527579a66a2da8ba04192acf629d7f168f19cabbb0fb2181ac: cannot remove container 46490b92bc2ab1527579a66a2da8ba04192acf629d7f168f19cabbb0fb2181ac as it could not be stopped: container 46490b92bc2ab1527579a66a2da8ba04192acf629d7f168f19cabbb0fb2181ac did not die within timeout
~ (aws:rean-gov-sd)(kc)$ runc list
ID                                                                 PID         STATUS      BUNDLE                                                                                                                                  CREATED                          OWNER
46490b92bc2ab1527579a66a2da8ba04192acf629d7f168f19cabbb0fb2181ac   14707       running     /home/frezbo/.local/share/containers/storage/vfs-containers/46490b92bc2ab1527579a66a2da8ba04192acf629d7f168f19cabbb0fb2181ac/userdata   2018-08-24T14:22:24.909422438Z   frezbo
~ (aws:rean-gov-sd)(kc)$ runc kill 46490b92bc2ab1527579a66a2da8ba04192acf629d7f168f19cabbb0fb2181ac KILL
~ (aws:rean-gov-sd)(kc)$ podman rm -f 46490b92bc2ab1527579a66a2da8ba04192acf629d7f168f19cabbb0fb2181ac

46490b92bc2ab1527579a66a2da8ba04192acf629d7f168f19cabbb0fb2181ac
~ (aws:rean-gov-sd)(kc)$ 
~ (aws:rean-gov-sd)(kc)$ 

```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
~ (aws:rean-gov-sd)(kc)$ podman --version
podman version 0.8.3
~ (aws:rean-gov-sd)(kc)$
```

**Output of `podman info`:**

```
~ (aws:rean-gov-sd)(kc)$ podman info
host:
  Conmon:
    package: podman-0.8.3-1.git9d09a4d.fc28.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 1.12.0-dev, commit: c72f69ce3f3fd7e46dced78dd99b67ab5441441e-dirty'
  MemFree: 3568762880
  MemTotal: 16724082688
  OCIRuntime:
    package: runc-1.0.0-46.dev.gitb4e2ecb.fc28.x86_64
    path: /usr/bin/runc
    version: 'runc version spec: 1.0.0'
  SwapFree: 0
  SwapTotal: 0
  arch: amd64
  cpus: 4
  hostname: frezbo
  kernel: 4.17.14-202.fc28.x86_64
  os: linux
  uptime: 6h 44m 55.15s (Approximately 0.25 days)
insecure registries:
  registries: []
registries:
  registries:
  - docker.io
  - registry.fedoraproject.org
  - quay.io
  - registry.access.redhat.com
  - registry.centos.org
store:
  ContainerStore:
    number: 36
  GraphDriverName: vfs
  GraphOptions: []
  GraphRoot: /home/frezbo/.local/share/containers/storage
  GraphStatus: {}
  ImageStore:
    number: 2
  RunRoot: /run/user/1000/run

~ (aws:rean-gov-sd)(kc)$
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**OS Info:**
```
~ (aws:rean-gov-sd)(kc)$ uname -a
Linux frezbo 4.17.14-202.fc28.x86_64 #1 SMP Wed Aug 15 12:29:25 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
~ (aws:rean-gov-sd)(kc)$
```

**Comment by mheon** [comment]
@giuseppe PTAL

**Comment by mheon** [comment]
Might be fixed by #1331

**Comment by giuseppe** [comment]
I've added a patch to #1331 to address the "rm -f" case.  I am not convinced if the fix should be in podman or in runc, as "kill --all" doesn't work when there is not a new cgroups (as it is the case with rootless containers).

/cc @AkihiroSuda
